### PR TITLE
Improve `@babel/runtime` esm stability

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            |  Maybe fixes https://github.com/babel/babel/issues/12861, but there are not enough info to tell it for sure.
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

We didn't get any bug report about this because technically it works, but this PR introduces two improvements that make `@babel/runtime` better when using ESM helpers with older tools like Webpack 4:
- It removes the warning about `useESModules`. This option is still necessary if you want to force Webpack 4 to use esm helpers
- It uses relative imports to load ESM dependencies of ESM helpers. Otherwise webpack 4 would resolve the CJS version, because it doesn't read export maps in `package.json` (we don't need to rewrite imports to cjs polyfills, since they are cjs anyway). Fixes https://github.com/babel/babel/issues/12881#issuecomment-785254057

I also added a CI step to check that `@babel/runtime` works with different Node.js versions and bundlers.

### CODE REVIEW

**Comment by JLHwung** [comment]
If we don't get any emergent bug reports, can you add runtime consumer e2e test in this PR? So we are confident that it does not break some bundler/engine versions.

**Comment by nicolo-ribaudo** [comment]
Yes

File: test/runtime-integration/src/main.mjs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import "./import-auto.mjs";
+import "./import-esm.mjs";
+import "./require-auto.cjs";
```
Comment (JLHwung) [inline test/runtime-integration/src/main.mjs:3]: Can you add a case on `./require-esm.cjs`?

File: test/runtime-integration/webpack-5/package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "@babel-internal/runtime-integration-webpack-5",
+  "private": true,
+  "exports": "./index.js",
+  "devDependencies": {
+    "@babel/runtime": "workspace:*",
```
Comment (JLHwung) [inline test/runtime-integration/webpack-5/package.json:6]: Can you duplicate cases to `@babel/runtime-corejs3`? We have extra core-js imports in that package.

**Review by JLHwung — APPROVED** [review]
LGTM with comments. Although we haven't received any bug reports on Rollup, should we add Rollup integration test?

File: test/runtime-integration/src/main.mjs (hunk 1 above)
Comment (nicolo-ribaudo) [reply test/runtime-integration/src/main.mjs:3]: Should it work or throw?

Comment (JLHwung) [reply test/runtime-integration/src/main.mjs:3]: It should throw on Node. But other bundlers, especially legacy versions may have different behaviour.

**Comment by nicolo-ribaudo** [comment]
Do we also need to test other bundlers, or do we trust that modern bundlers will do the correct thing?

**Review by JLHwung — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
(The error messages are different in different node versions, I'm checking all of them)

**Review by kaicataldo — APPROVED** [review]

**Pull request merged**